### PR TITLE
Add the groups to roles

### PR DESCRIPTION
**Pull request by drapergeek**

### CODE REVIEW

File: lib/cog_api/fake/roles.ex [hunk 1]
```diff
@@ -75,4 +75,11 @@ defmodule CogApi.Fake.Roles do
   defp permissions_match?(%Permission{} = permission, %Permission{} = match) do
     permission.namespace == match.namespace && permission.name == match.name
   end
+
+  defp prepare_role(role) do
```
Comment (jsteiner) [inline lib/cog_api/fake/roles.ex]: I think I'm missing something. Why isn't this solved already by adding `groups` to the `associations` keyword list?

Comment (drapergeek) [inline lib/cog_api/fake/roles.ex]: This is the wrong side of the association. We don't store the `group_ids` here, this has to look those up instead.

This is more like the `has_one` side of the equation if that helps.

Comment (jsteiner) [inline lib/cog_api/fake/roles.ex]: Got it.

**Comment by jsteiner** [comment]
👍

**Pull request merged**